src/eventshub/src/utilities/algoliaFiltersData.ts

```diff
@@ -0,0 +1,66 @@
+import { SearchIndex } from 'algoliasearch';
+import searchClient from './algoliaConfig'; 
+
+export interface AlgoliaHit {
+  objectID: string;
+  name: string;
+  description: string;
+}
+
+const index: SearchIndex = searchClient.initIndex('SERENE ALL - appEb3LGlZS9OfNrK - Relationships');
+
+export const fetchAlgoliaData = async (): Promise<AlgoliaHit[]> => {
+  try {
+    const { hits } = await index.search<AlgoliaHit>('', { hitsPerPage: 1000 });
+    return hits;
+  } catch (error) {
+    console.error('Error fetching data from Algolia:', error);
+    return [];
+  }
+};
+
+export interface DataItem {
+  event_feature: string;
+  feature_group: string;
+  'policies:_logistics_categories_column': string[];
+  'vendors:_service_type': string[];
+  'venues:_space_type': string[];
+}
+
+export interface EventFeatureGroupItem {
+  event_feature: string;
+  venues: string[];
+  vendors: string[];
+  policies: string[];
+}
+
+export interface GroupedDataItem {
+  feature_group: string;
+  event_feature_group: EventFeatureGroupItem[];
+}
+
+export const restructureData = (data: DataItem[]): GroupedDataItem[] => {
+  const uniqueFeatureGroups: string[] = Array.from(new Set(data.map(item => item.feature_group)));
+
+  return uniqueFeatureGroups.map((group) => {
+    const itemsInGroup: DataItem[] = data.filter(item => item.feature_group === group);
+
+    const eventFeatureGroup: EventFeatureGroupItem[] = itemsInGroup.map(item => ({
+      event_feature: item.event_feature,
+      venues: item['venues:_space_type'],
+      vendors: item['vendors:_service_type'],
+      policies: item['policies:_logistics_categories_column'],
+    }));
+
+    return {
+      feature_group: group,
+      event_feature_group: eventFeatureGroup,
+    };
+  });
+}
+
+export const algoliaFilterData = async (): Promise<GroupedDataItem[]> => {
+  const algoliaData = await fetchAlgoliaData();
+  const filterData = restructureData(algoliaData as unknown as DataItem[]);
+  return filterData;
```

Comment:
```[
  {
    "feature_group": "Food, Drinks, or Items",
    "event_feature_group": [
      {
        "event_feature": "Food Truck at Event",
        "venues": "Uncategorized, Field(s), Outdoor Space",
        "vendors": ["Food/Water Trucks"],
        "policies": ["Food", "Permits", "Catering and Alcohol"]
      },
      {
        "event_feature": "Alcohol",
        "venues": "Uncategorized",
        "vendors": ["Alcohol Service"],
        "policies": ["Alcohol", "Catering and Alcohol", "Licensing", "Permits"]
      },
      {
        "event_feature": "Selling Items at Event",
        "venues": "Uncategorized",
        "vendors": ["Design/Print", "Merchandise", "Graphic Designer"],
        "policies": ["Retail", "Budgeting and Purchasing"]
      },
      {
        "event_feature": "Tabling at Event",
        "venues": "Uncategorized",
        "vendors": ["Merchandise", "Design/Print", "Graphic Designer"],
        "policies": ["Retail"]
      },
      {
        "event_feature": "Swag at Event",
        "venues": "Uncategorized, Conference Center, Performance Space",
        "vendors": ["Merchandise", "Design/Print", "Graphic Designer"],
        "policies": [
          "Retail",
          "Promotion",
          "Budgeting and Purchasing",
          "Marketing"
        ]
      },
      {
        "event_feature": "Food",
        "venues": "Uncategorized",
        "vendors": ["Catering/Food Service", "Food/Water Trucks"],
        "policies": ["Food", "Catering and Alcohol"]
      }
    ]
  },
  {
    "feature_group": "Other",
    "event_feature_group": [
      {
        "event_feature": "Tenting",
        "venues": "Outdoor Space, Field(s), Uncategorized",
        "vendors": ["Stanford Support", "Equipment Rentals"],
        "policies": ["Outdoor", "Use of Campus Space"]
      },
      {
        "event_feature": "Speaker/Presentation",
        "venues": "Uncategorized",
        "vendors": ["Photography/Videography"],
        "policies": ["Special Visitor Support", "Special Events"]
      },
      {
        "event_feature": "Involves Minors",
        "policies": ["Risk Management and Liability", "Security and Safety"]
      },
      {
        "event_feature": "Amplified Sound",
        "venues": "Performance Hall, Performance Space, Amphitheater, Uncategorized",
        "vendors": [
          "Stanford Support",
          "Musicians/Entertainers",
          "Entertainment Rentals",
          "Equipment Rentals"
        ],
        "policies": [
          "Use of Campus Space",
          "Permits",
          "Security and Safety",
          "Performance"
        ]
      },
      {
        "event_feature": "Team Activity",
        "venues": "Outdoor Space",
        "vendors": [
          "Alcohol Service",
          "Catering/Food Service",
          "Activity",
          "Party/Game/Entertainment Rentals",
          "Entertainment Rentals",
          "Transportation"
        ]
      },
      {
        "event_feature": "Fundraising",
        "venues": "Uncategorized",
        "vendors": [
          "Catering/Food Service",
          "Graphic Designer",
          "Equipment Rentals"
        ],
        "policies": ["Fundraising/Charity", "Budgeting and Purchasing"]
      },
      {
        "event_feature": "Ticketing, Registration, or Charging Admission",
        "venues": "Auditorium/Auditoria, Performance Hall, Performance Space",
        "vendors": ["Stanford Support"],
        "policies": [
          "Ticketing",
          "Date Selection",
          "Marketing",
          "Use of Campus Space",
          "Budgeting and Purchasing"
        ]
      }
    ]
  },
  {
    "feature_group": "Size",
    "event_feature_group": [
      {
        "event_feature": "Large Venue (500-1,000)",
        "venues": "Ballroom, Uncategorized"
      },
      {
        "event_feature": "X-Large Venue (1,000+)",
        "venues": "Amphitheater, Auditorium/Auditoria, Stadiums/Stadium, Field(s), Uncategorized"
      },
      {
        "event_feature": "Medium Venue (100-500)",
        "venues": "Uncategorized"
      },
      {
        "event_feature": "Small Venue (Less than 100)",
        "venues": "Classroom(s), Meeting Room(s), Studio(s), Patio(s), Lounge"
      }
    ]
  },
  {
    "feature_group": "Support",
    "event_feature_group": [
      {
        "event_feature": "Needs Additional Equipment ",
        "venues": "Uncategorized",
        "vendors": [
          "Equipment Rentals",
          "Stanford Support",
          "Entertainment Rentals",
          "Party/Game/Entertainment Rentals"
        ],
        "policies": ["Permits", "Budgeting and Purchasing"]
      },
      {
        "event_feature": "Needs Lodging Support ",
        "venues": "Uncategorized",
        "vendors": ["Stanford Support"],
        "policies": ["Hospitality"]
      },
      {
        "event_feature": "Involves Photography or Videography",
        "venues": "Uncategorized",
        "vendors": ["Photography/Videography"],
        "policies": ["Media"]
      },
      {
        "event_feature": "Needs Volunteer/Staff Support",
        "venues": "Uncategorized",
        "vendors": ["Stanford Support"],
        "policies": ["Permits", "Use of Campus Space"]
      },
      {
        "event_feature": "A/V Support Needed",
        "venues": "Uncategorized",
        "vendors": ["Stanford Support", "Equipment Rentals"],
        "policies": ["Media", "Use of Campus Space"]
      },
      {
        "event_feature": "Needs Security Support",
        "venues": "Uncategorized",
        "vendors": ["Stanford Support"],
        "policies": ["Security and Safety", "Risk Management and Liability"]
      },
      {
        "event_feature": "Needs Parking or Transportation Support",
        "venues": "Uncategorized",
        "vendors": ["Transportation", "Stanford Support", "Food/Water Trucks"],
        "policies": ["Transportation", "Parking"]
      }
    ]
  },
  {
    "feature_group": "Indoor/Outdoor/Mixed",
    "event_feature_group": [
      {
        "event_feature": "Outdoor Venue",
        "venues": "Outdoor Space, Terrace, Amphitheater, Sculpture Garden, Patio(s), Uncategorized, Stadiums/Stadium, Plaza, Garden(s), Cafe, Performance Space, Field(s), Exercise Space",
        "vendors": [
          "Party/Game/Entertainment Rentals",
          "Equipment Rentals",
          "Entertainment Rentals"
        ],
        "policies": ["Outdoor", "Use of Campus Space"]
      },
      {
        "event_feature": "Indoor Venue",
        "venues": "Uncategorized, Auditorium/Auditoria, Seminar Room, Ballroom, Classroom(s), Conference Room, Dining Room, Living Room, Meeting Room(s), Hall, Lecture Hall, Performance Hall, Reception Hall, Studio(s), Church, Lounge, Foyer, Gallery, Indoor Space, Lobby",
        "vendors": ["Stanford Support"],
        "policies": ["Use of Campus Space"]
      }
    ]
  }
]
```

@sherakama thoughts on this filter data restructure? 🤔

Reply:
I'll add it to the set of questions for the airtble content authors

Reply:
It's looking good. I've added a weight field in the data and put the default sorting by it.

Reply:
Thanks @sherakama! I've updated the restructure data function to sort the filter items by the weight defined in the Airtable 😃 

I'll be moving onto the filter state management and algolia queries today unless there are any additional adjustments to the restructured data we'd like to make 🤔